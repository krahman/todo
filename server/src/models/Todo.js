const mongoose = require('mongoose');

const Schema = mongoose.Schema;
const TodoSchema = new Schema({
  activity: String,
  completed: Boolean,
});

const Todo = mongoose.model('Todo', TodoSchema);
module.exports = Todo;
