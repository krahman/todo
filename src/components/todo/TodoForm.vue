<template>
  <el-container class="todo-form-container">
    <el-main class="todo-form">
      <el-form :inline="true" class="todo-form-inline" @submit.prevent.native="addTodo">
        <el-row>
          <el-col :span="16">
            <el-input placeholder="Add new task here" v-model="todo.activity"
                      class="todo-form-input" ref="activity" autofocus="true"></el-input>
          </el-col>
          <el-col :span="2">
            <el-button type="primary" @click="addTodo" class="todo-form-button"
                       :disabled="!todo.activity">Add</el-button>
          </el-col>
        </el-row>
      </el-form>
    </el-main>
  </el-container>
</template>

<script src="../../controllers/TodoForm.js"></script>

<style scoped>
  @import "../../assets/styles/todo-form.css";
</style>
