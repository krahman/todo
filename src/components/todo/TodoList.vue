<template>
  <el-container class="todo-list-container">
    <el-main v-if="isEmpty" class="todo-list-empty">
      <div v-if="!isLoading">Nothing todo today, Yayy!!!</div>
    </el-main>
    <el-main v-if="!isEmpty" class="todo-list">
      <todo-item v-for="todo in todos" v-bind:key="todo._id" :todos="todos" :todo="todo">
      </todo-item>
    </el-main>
  </el-container>
</template>

<script src="../../controllers/TodoList.js"></script>

<style scoped>
  @import "../../assets/styles/todo-list.css";
</style>
