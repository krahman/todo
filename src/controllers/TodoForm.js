import TodoService from '@/services/TodoService';
import { ADD_TODO } from '@/stores/mutation-types';

export default {
  name: 'TodoForm',
  computed: {
    todo() {
      return this.$store.state.todo;
    },
  },
  methods: {
    async addTodo() {
      if (!this.todo.activity) {
        this.$message({
          type: 'error',
          message: 'Task can not be empty.',
        });
      } else {
        const response = await TodoService.addTodo({
          activity: this.todo.activity,
          completed: false,
        });
        this.$store.commit(ADD_TODO, response.data);
      }
      this.$refs.activity.focus();
    },
  },
};
