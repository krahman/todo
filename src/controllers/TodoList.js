import TodoService from '@/services/TodoService';
import TodoItem from '@/components/todo/TodoItem';
import { FETCH_TODO } from '@/stores/action-types';

export default {
  name: 'TodoList',
  data() {
    return {
      isLoading: false,
    };
  },
  components: {
    TodoItem,
  },
  computed: {
    todos() {
      return this.$store.state.todos;
    },
    isEmpty() {
      return this.$store.state.todos.length < 1;
    },
  },
  mounted() {
    this.getTodos();
  },
  methods: {
    async getTodos() {
      this.isLoading = true;
      const response = await TodoService.fetchTodos();
      this.$store.commit(FETCH_TODO, response.data);
      this.isLoading = false;
    },
  },
};
