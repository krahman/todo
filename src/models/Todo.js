export default class Todo {
  constructor({ activity = null, completed = false }) {
    this.activity = activity;
    this.completed = completed;
  }
}

export const addTodo = (todo) => {
  const { activity, completed } = todo;

  return Object.freeze(new Todo({ activity, completed }));
};
