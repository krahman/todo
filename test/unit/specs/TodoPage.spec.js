import Todo from '@/components/todo/TodoPage';
import TodoForm from '@/components/todo/TodoForm';
import TodoList from '@/components/todo/TodoList';
import { createLocalVue, shallowMount } from '@vue/test-utils';
import { shallow } from 'avoriaz';
import Vuex from 'vuex';

const localVue = createLocalVue();
localVue.use(Vuex);

describe('Todo.vue', () => {
  let store;

  beforeEach(() => {
    store = new Vuex.Store({
      state: {
        todos: ['', ''],
        todo: {
          activity: '',
          completed: false,
        },
      },
    });
  });

  it('matches Todo snapshot', () => {
    const wrapper = shallowMount(Todo, {});
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should contains TodoForm', () => {
    const wrapper = shallow(Todo);
    expect(wrapper.contains(TodoForm)).toEqual(true);
  });

  it('matches TodoForm snapshot', () => {
    const wrapper = shallowMount(TodoForm, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should contains TodoList', () => {
    const wrapper = shallow(Todo);
    expect(wrapper.contains(TodoList)).toEqual(true);
  });

  it('matches TodoList snapshot', () => {
    const wrapper = shallowMount(TodoList, { store, localVue });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
