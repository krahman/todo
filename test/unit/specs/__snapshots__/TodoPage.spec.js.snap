// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo.vue matches Todo snapshot 1`] = `
<el-container class="todo-page">
    <el-main>
        <todoform-stub></todoform-stub>
        <todolist-stub></todolist-stub>
    </el-main>
</el-container>
`;

exports[`Todo.vue matches TodoForm snapshot 1`] = `
<el-container class="todo-form-container">
    <el-main class="todo-form">
        <el-form inline="true" class="todo-form-inline">
            <el-row>
                <el-col span="16">
                    <el-input placeholder="Add new task here" autofocus="true" class="todo-form-input"></el-input>
                </el-col>
                <el-col span="2">
                    <el-button type="primary" disabled="true" class="todo-form-button">Add</el-button>
                </el-col>
            </el-row>
        </el-form>
    </el-main>
</el-container>
`;

exports[`Todo.vue matches TodoList snapshot 1`] = `
<el-container class="todo-list-container">
    <!---->
    <el-main class="todo-list">
        <todoitem-stub todos="," todo=""></todoitem-stub>
        <todoitem-stub todos="," todo=""></todoitem-stub>
    </el-main>
</el-container>
`;
